src/metax_api/models/catalog_record.py

```diff
@@ -1221,6 +1228,11 @@ def _post_create_operations(self):
                 self.add_post_request_callable(DataciteDOIUpdate(self, self.research_dataset['preferred_identifier'],
                                                                 'create'))
 
+            if self.use_doi_for_draft is True:
+                self._validate_cr_against_datacite_schema()
+                self.add_post_request_callable(DataciteDOIUpdate(self, self.research_dataset['preferred_identifier'],
+                                                                'create'))
+
```

Comment:
Does exactly the same thing as above if-clause so these could be combined

Reply:
Ok combined to earlier conditional with an 'or'